pipeline/transformer/transformer_common/src/main/scala/weco/pipeline/transformer/identifiers/SourceIdentifierValidation.scala

```diff
@@ -56,7 +56,8 @@ object SourceIdentifierValidation {
      * see: `https://search.wellcomelibrary.org/iii/encore/record/C__Rb1971204?marcData=Y`
      */
     private def tryParseUUID(str: String): Try[UUID] = Try(UUID.fromString(str))
-
+    private def isValidMetsId(str: String): Boolean =
+      sierraSystemNumber.toPredicate(str) || tryParseUUID(str).isSuccess
```

Comment:
Feels a little strange that we need to do this ... did we think about having a sierra identifier type and an archivematica identifier type? I'm guessing there is something awkward about the adapter data model that makes this difficult?

Reply:
I'll put some commentary in about it.

That relates to [this ramble](https://wellcome.slack.com/archives/C02ANCYL90E/p1705060396703809?thread_ts=1705058777.597079&cid=C02ANCYL90E).

I think it's perfectly possible to have two distinct METS ID types here. The METS Transformer already knows whether it's a b number or a UUID because of where it comes from.  Similarly, the CALM transformer will already know which type it is when creating the mergeCandidate for the same reason. However, I don't think it's a valuable distinction.

It boils down to this - 
1. The existing behaviour works this way because:
    * Goobi METS files only have the Sierra B number as their identifier
    * That B Number is also the identifier for the Sierra record it corresponds to
    * The identifier for a document must be unique across the whole system
    * Therefore, we put these two identical identifiers into two distinct namespaces
2. The behaviour I have added simply puts the METS UUID into the METS database because:
    * It comes from METS, and we have an existing METS namespace
    * That it is from METS is the only practical/important distinction here. This is used in the mergeCandidate code
    * The namespace must reflect the source of the id, otherwise the b number problem could arise
    * I think the only thing that would be gained from having two distinct METS id types, would be to simplify this one line at the expense of (admittedly trivial) complexity elsewhere.
    * The existing METS type is already named with the unmarked form METS, and not SierraMets or METSBnumber, meaning that METSUUID would be an unfairly marked form.  The existing name cannot bet changed without significant messing about in the ID Minter.